packages/snaps-controllers/src/interface/utils.ts

```diff
@@ -285,6 +424,47 @@ export function getAssetSelectorStateValue(
   };
 }
 
+/**
+ * Get the state value for an account selector.
+ *
+ * @param element - The account selector element.
+ * @param elementDataGetters - Data getters for the element.
+ * @param elementDataGetters.getAccountByAddress - A function to get an account by address.
+ * @param elementDataGetters.setSelectedAccount - A function to set the selected account in the client.
+ * @param elementDataGetters.snapOwnsAccount - A function to check if the snap owns the account.
+ * @returns The state value for the account selector.
+ */
+export function getAccountSelectorStateValue(
+  element: AccountSelectorElement,
+  {
+    getAccountByAddress,
+    setSelectedAccount,
+    snapOwnsAccount,
+  }: ElementDataGetters,
+) {
+  const { value, switchGlobalAccount, hideExternalAccounts } = element.props;
+
+  if (!value) {
+    return null;
+  }
+
+  const account = getAccountByAddress(value);
+
+  if (!account) {
+    return null;
+  }
+
+  if (hideExternalAccounts && !snapOwnsAccount(account)) {
+    return null;
+  }
+
+  if (switchGlobalAccount) {
+    setSelectedAccount(account.id);
```

Comment:
Seems strange for a getter to have a side-effect 🤔

Reply:
It is indeed but it's not prettier to have it outside of the getter 😅 I can move it if you prefer

Reply:
It would live in the component but we need to be able to switch the account here for the default state ? 🤔

Reply:
Default state is still propagated to the component which could then switch the account selected, no?

Reply:
That could work yes !